esmvalcore/preprocessor/_area.py

```diff
@@ -156,19 +161,32 @@ def tile_grid_areas(cube, fx_files):
                 continue
             logger.info('Attempting to load %s from file: %s', key, fx_file)
             fx_cube = iris.load_cube(fx_file)
-
             grid_areas = fx_cube.core_data()
-            if cube.ndim == 4 and grid_areas.ndim == 2:
-                grid_areas = da.tile(grid_areas,
-                                     [cube.shape[0], cube.shape[1], 1, 1])
-            elif cube.ndim == 4 and grid_areas.ndim == 3:
-                grid_areas = da.tile(grid_areas, [cube.shape[0], 1, 1, 1])
-            elif cube.ndim == 3 and grid_areas.ndim == 2:
-                grid_areas = da.tile(grid_areas, [cube.shape[0], 1, 1])
-            else:
-                raise ValueError('Grid and dataset number of dimensions not '
-                                 'recognised: {} and {}.'
-                                 ''.format(cube.ndim, grid_areas.ndim))
+    else:
+        return None
+
+    if grid_areas.shape != cube.shape[-2:]:
+        raise ValueError('Fx area {} and dataset {} shapes do not match.'
+                         ''.format(grid_areas.shape, cube.shape))
+
+    if cube.ndim == grid_areas.ndim:
+        return grid_areas
+
+    # Use dash.array.stack to tile areacello.
+    elif cube.ndim == 4 and grid_areas.ndim == 2:
+        for shape in [1, 0]:
+            grida = [grid_areas for itr in range(cube.shape[shape])]
+            grid_areas = da.stack(grida, axis=0)
+    elif cube.ndim == 4 and grid_areas.ndim == 3:
+        grida = [grid_areas for itr in range(cube.shape[0])]
+        grid_areas = da.stack(grida, axis=0)
+    elif cube.ndim == 3 and grid_areas.ndim == 2:
+        grida = [grid_areas for itr in range(cube.shape[0])]
+        grid_areas = da.stack(grida, axis=0)
+    else:
+        raise ValueError('Grid and dataset number of dimensions not '
+                         'recognised: {} and {}.'
+                         ''.format(cube.ndim, grid_areas.ndim))
```

Comment:
Could [`broadcast_to`](https://docs.dask.org/en/latest/array-api.html#dask.array.broadcast_to) be used to simplify this block of `if`s?